src/core/schema/src/main/python/TypeConversionUtils.py

```diff
@@ -28,16 +30,20 @@ def complexTypeConverter(name, value, cache):
         _java_obj:
     """
     cache[name]=value
-    if isinstance(value, list):
-        java_value=[]
-        for v in value:
-            if hasattr(v, "_transfer_params_to_java"):
-                v._transfer_params_to_java()
-            java_value.append(v._java_obj)
-        return java_value
-    if hasattr(value, "_transfer_params_to_java"):
-        value._transfer_params_to_java()
-    if hasattr(value, "_java_obj"):
-        return value._java_obj
-    else:
-        return value._to_java()
+    try:
+        if isinstance(value, list):
```

Comment:
OK, do you really need ``value`` to be a list or can it be any iterable object? I generally prefer to not check for concrete instances since it's not really idiomatic in a duck typed language.